Meta/CMake/libweb_data.cmake

```diff
@@ -0,0 +1,293 @@
+        )
+        # FIXME: Instead of requiring a manual declaration of iterable wrappers, we should ask WrapperGenerator if it's iterable
+        if(LIBWEB_WRAPPER_ITERABLE)
+            list(APPEND BINDINGS_SOURCES
+                "Bindings/${basename}IteratorWrapper.h"
+                "Bindings/${basename}IteratorWrapper.cpp"
+                "Bindings/${basename}IteratorPrototype.h"
+                "Bindings/${basename}IteratorPrototype.cpp"
+            )
+            list(APPEND BINDINGS_TYPES
+                iterator-header
+                iterator-implementation
+                iterator-prototype-header
+                iterator-prototype-implementation
+            )
+        endif()
+        target_sources(LibWeb PRIVATE ${BINDINGS_SOURCES})
+        # FIXME: cmake_minimum_required(3.17) for ZIP_LISTS
+        list(LENGTH BINDINGS_SOURCES num_bindings)
+        math(EXPR bindings_end "${num_bindings} - 1")
+        foreach(iter RANGE "${bindings_end}")
+            list(GET BINDINGS_SOURCES ${iter} bindings_src)
+            list(GET BINDINGS_TYPES ${iter} bindings_type)
+            get_property(include_paths DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
+            list(TRANSFORM include_paths PREPEND -i)
+            add_custom_command(
+                OUTPUT "${bindings_src}"
+                COMMAND "$<TARGET_FILE:Lagom::WrapperGenerator>" "--${bindings_type}" ${include_paths} "${CMAKE_CURRENT_SOURCE_DIR}/${class}.idl" "${CMAKE_CURRENT_SOURCE_DIR}/" > "${bindings_src}.tmp"
+                COMMAND "${CMAKE_COMMAND}" -E copy_if_different "${bindings_src}.tmp" "${bindings_src}"
+                COMMAND "${CMAKE_COMMAND}" -E remove "${bindings_src}.tmp"
+                VERBATIM
+                DEPENDS Lagom::WrapperGenerator
+                MAIN_DEPENDENCY ${class}.idl
+            )
+        endforeach()
+        add_custom_target(generate_${basename}Wrapper.h DEPENDS Bindings/${basename}Wrapper.h)
+        add_dependencies(all_generated generate_${basename}Wrapper.h)
+        add_custom_target(generate_${basename}Wrapper.cpp DEPENDS Bindings/${basename}Wrapper.cpp)
+        add_dependencies(all_generated generate_${basename}Wrapper.cpp)
+        add_custom_target(generate_${basename}Constructor.h DEPENDS Bindings/${basename}Constructor.h)
+        add_dependencies(all_generated generate_${basename}Constructor.h)
+        add_custom_target(generate_${basename}Constructor.cpp DEPENDS Bindings/${basename}Constructor.cpp)
+        add_dependencies(all_generated generate_${basename}Constructor.cpp)
+        add_custom_target(generate_${basename}Prototype.h DEPENDS Bindings/${basename}Prototype.h)
+        add_dependencies(all_generated generate_${basename}Prototype.h)
+        add_custom_target(generate_${basename}Prototype.cpp DEPENDS Bindings/${basename}Prototype.cpp)
+        add_dependencies(all_generated generate_${basename}Prototype.cpp)
+    endfunction()
+
+    libweb_js_wrapper(Crypto/Crypto)
```

Comment:
It feels really weird to have these specific calls out in the CMake helper directory, now when you add a new class that needs libweb wrappers you have to edit two files :( .... might be time to make Lagom/CMakeLists.txt actually just `include_directory(Userland)` 🤔

Reply:
I created a new `idl_files.cmake` file in the LibWeb folder to solve this - it's still less normal than before, needing to use a specific file instead of `CMakeLists.txt` like before, but at least there's no need to look into another folder anymore!